src/initial_condition.jl

```diff
@@ -1,25 +1,84 @@
 # Define initial condition
-function init_1d_column(::Type{FT}, params, z) where {FT}
-    # physics parameters
-    # R_d::FT = CLIMAParameters.Planet.R_d(params)
+function init_condition(params, z)
+
+    FT = Float64
 
     z_0::FT = 0.0
     z_1::FT = 740.0
     z_2::FT = 3260.0
     qv_0::FT = 0.016
     qv_1::FT = 0.0138
     qv_2::FT = 0.0024
-    θ_0::FT = 279.9
-    θ_1::FT = 279.9
+    θ_0::FT = 297.9
+    θ_1::FT = 297.9
     θ_2::FT = 312.66
 
-    # density, pressure, etc...
+    # profile of water vapour specific humidity (TODO - or is it mixing ratio?)
+    qv = z < z_1 ? qv_0 + (qv_1 - qv_0)/(z_1 - z_0) * z : qv_1 + (qv_2 - qv_1)/(z_2 - z_1) * z
 
-    # potential temperature
+    # profile of potential temperature
     θ = z < z_1 ? θ_0 : θ_1 + (θ_2 - θ_1)/(z_2 - z_1) * z
 
-    # water vapour specific humidity (TODO - or is it mixing ratio?)
-    qv = z < z_1 ? qv_0 + (qv_1 - qv_0)/(z_1 - z_0) * z : qv_1 + (qv_2 - qv_1)/(z_2 - z_1) * z
+    # density at the surface
+    p_0::FT = 100200.0
+    q_0 = TD.PhasePartition(qv_0, 0.0, 0.0)
+    T_0 = θ_0 * TD.exner_given_pressure(params, p_0,  q_0)
+    ρ_0 = TD.air_density(params, T_0, p_0, q_0)
+
+    return(qv = qv, θ = θ, ρ_0 = ρ_0, z_0 = z_0, z_2 = z_2)
+end
+
+function dρ_dz!(ρ, params, z)
+
+    FT = Float64 #eltype(CP.Planet.grav(params))
+
+    # initial profiles
+    init = init_condition(params, z)
+    θ::FT = init.θ
+    qv::FT = init.qv
+
+    q = TD.PhasePartition(qv, 0.0, 0.0)
+
+    # constants
+    g::FT = CP.Planet.grav(params)
+    cp_m::FT = TD.cp_m(params, q)
+    R_m::FT = TD.gas_constant_air(params, q)
+
+    T::FT = θ * (ρ * θ / CP.Planet.MSLP(params) * R_m)^((R_m / cp_m) / (1 - R_m / cp_m))
+
+    return g / T * ρ * (R_m / cp_m - 1) / R_m
+end
+
+function ρ_ivp(params)
+
+    FT = Float64 #eltype(CP.Planet.grav(params))
+
+    init_surface = init_condition(params, 0.0)
+
+    ρ_0 = init_surface.ρ_0
+    z_0 = init_surface.z_0
+    z_max = init_surface.z_2
+
+    z_span = (z_0, z_max)
+    prob = ODEProblem(dρ_dz!, ρ_0, z_span, params)
+    sol = solve(prob, Tsit5(), reltol=1e-8, abstol=1e-8)
+
+    return sol
+end
+
+function init_1d_column(params, ρ_profile, z)
+
+    FT = Float64
+
```

Comment:
Maybe cache the init_condition rather than computing it twice?

```
init_column = init_conditions(params,z)
θ_liq_ice::FT = FT(init_column.θ)
q_tot::FT = FT(init_column.qv)
```

Reply:
I think it's fine. This only happens in the initialisation. So it shouldn't affect us much. On the other hand adding it to cache would add two additional profiles to store during the whole simulation, even though we only need them at initialisation.

tbh, with a model this small and simple it probably doesn't matter either way. I'll ask around what is considered the better practice